guides/micronaut-data-jdbc-repository/groovy/src/test/groovy/example/micronaut/GenreControllerSpec.groovy

```diff
@@ -0,0 +1,145 @@
+package example.micronaut
+
+import example.micronaut.domain.Genre
+import io.micronaut.http.client.HttpClient
+import io.micronaut.http.client.annotation.Client
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.HttpStatus
+import io.micronaut.http.HttpHeaders
+import io.micronaut.core.type.Argument
+import io.micronaut.http.client.exceptions.HttpClientResponseException
+import io.micronaut.test.extensions.spock.annotation.MicronautTest
+import spock.lang.Specification
+import spock.lang.Stepwise
+import spock.lang.Unroll
+
+import jakarta.inject.Inject
+
+@MicronautTest //<1>
+@Stepwise
```

Comment:
I personally consider `@Stepwise` a spock smell. I would prefer a single test which does not leave any side effect in the database as in the java guide. 

Excerpt From: Rob Fletcher [Spock: Up and Running](https://www.oreilly.com/library/view/spock-up-and/9781491923283/)

> However, on very rare occasions it’s actually desirable to create a specification whose feature methods are dependent on one another and need to run in order, and whose changes to shared state are maintained.”

You can keep the nice test descriptions you wrote in the `when` labels. E.g. .

`when: 'verify get/read operation'`

Reply:
I've changed the test